### ISSUE TITLE
15.3.15 neue firmware signaturen

### ISSUE THREAD

**Issue by flemming-pr** [freifunk-flensburg/Voting#4]
Momentan haben wir in der s-1.1 4 signaturen von denen müssen 3 aus 4 richtig sein eine andere idee zwischen durch war das wir 4 aus sieben machen. 
mein jetziger vorschlag beinhaltet jedoch nur die key von mir sven und felix da ich momentan das gefuhl haben das wir die drei sind die sich am meisten mit der firmware auseiander setzten und und auch am besten auskennen. von daher würde ich es auf 2 aus 3 reduzieren viele andere communyties verfahren ebenso

**Comment by Wlanfr3ak** [comment]
Ich würde auch gerne mit meinem Key dort drinnen bleiben

**Comment by horstfffl** [comment]
ich bin für mehr keys anstatt weniger.
good_signatures =  3 von x
x--> soviele wie wollen
wer weiß wer in 3 monaten auf einmal keine zeit grad hat firmware zu testen/signen da ist es gut ein paar in reserve zu haben.
also felix und mein key neu rein
und bitte keine löschen.

**Comment by bigfootDK** [comment]
Ich fände eine ungerade anzahl firmware-signers am besten. von daher 3, 5 oder 7.

**Comment by flemming-pr** [comment]
Also bis jetzt sind wir Felix Sven ich Fabian und Horst willst du auch noch signieren? Denn haben wir 3 aus 5 ist finde ich gut und auch ausreichend

**Comment by horstfffl** [comment]
joa mein key +1
wollen dann rene und thorben nicht mehr signen ?
einfach so rauschmeißen ist uncool.

**Comment by spralexx** [comment]
doch, ich möchte weiterhin signen. hab auch grade meinen key geupdated

**Comment by foertel** [comment]
Es geht dabei ja nicht darum, wer alles signen will. Sondern darum, wer aktiv an der firmware arbeitet. Das sind im Moment Flemming und ich. Dann noch Rene und Sven dazu, weil die ab und zu testen un Feedback geben. (Zwei aus) Vier Keys ... mehr als genug (siehe auch andere Communities).

**Comment by horstfffl** [comment]
interesant das du so festlegst wer so "aktiv" ist 
ich halte mich jetz einfach auch mal ganz frech für aktiv.

"machen die andern communites auch so finde ich nen schwaches argument bzw garkeines"
zumnindest ohne ne genaue untersuchung warum das bei denen so ist.
könnte ja auch sein das die einfahc nicht mehr manpower haben.

**Comment by foertel** [comment]
Ich lege nicht fest. 

Ich äußere meine Meinung dazu. Und stimme entsprechend ab. Und Leute werden meiner Argumentation folgen oder eben nicht und auch dem entsprechend abstimmen. Und dann wird gemacht, was die meisten wollen.

Ein Hoch auf die Demokratie.

**Comment by horstfffl** [comment]
"Ein Hoch auf die Demokratie." !!
genau deswegen soviele signer wie möglich und mindestens 3 sollten signen müssen bevor ein update rausgeht. 

alles andere ist sonnst wohl Autokratie.

**Comment by S4ra** [comment]
Fassen wir mal zusammen:
Aktuell signen möchten und/oder können Felix, Sven, Flemming, Fabian, Horst, Rene, Thorben, also 7 Personen. 
<b>+1 soviele Signer wie möglich</b>

Und ein Vorschlag lautet: Wenn eine neue Version der Firmware rauskommt, sollte sie von mind. 3 der aufgezählten gesigned werden. 
<b>+1 mind. 3 Signer</b>

Und ein weiterer, zusätzlicher Vorschlag lautet: Wenn sie von mehr als 3 Personen gesigned wird, müssen mind. 3 Signaturen "richtig" sein.
@flemssound: Kam das denn schon mal vor, dass da ne gefälschte Signatur bei war oder ist das mehr ein theoretischer Fall?
<b>+1  3 von x müssen "richtig" sein</b>

**Comment by foertel** [comment]
Ich befürchte, dass nicht allen hier klar ist, wofür die Signatur gut ist.

Die Signatur bestätigt, dass die gebaute Firmware mit der zu bauenden Config übereinstimmt. Dass auf dem Weg "Firmware-Bäcker" -> "Node" die Firmware nicht von dritten manipuliert wurde. Es reicht hierfür die Signatur des Firmware-Bäckers.

Damit niemand dessen Key klaut und es damit einen einfachen Angriffs-Vektor gibt, baut man normalerweise eine zwei Signatur von jemandem ein, der auch am Bau der Firmware beteiligt ist. Zusätzliche Sicherheit.

Das hat nichts mit Demokratie und Kontrolle und Test-Protokollen zu tun. Sondern nur damit, dass gebaute Firmware nicht von dritten manipuliert wird.

Ich bleibe deswegen bei meinem Vorschlag: 2 aus 4.

**Comment by horstfffl** [comment]
ich finde die signaturen können neben dem authentifizieren auch sehr gut für die erhaltung eines demokratischen gleichgewichts ala https://de.wikipedia.org/wiki/Checks_and_Balances verwendet werden.
wir haben halt ansonnsten keine verbindlichen möglichkeiten das die community einfluss auf die firmware-bereitsteller hat.

es würden  in foertels beispiel 2 signaturen reichen und zugang zu dem speicherort der updatedateien reichen um die firmware zu verteilen. das hieße aber auch das zb wenn sven und ich die firmware bauen und sie auch selbst beide signieren können.. wir da ohne weitere kontrolle das ganze netz (-der_gateways) übernehmen könnten.

ich würde ja soein check and balances system für uns vorschlagen:

die community stimmt öffentlich auf github die änderungen an der firmware ab. 
die firmware-bauer bauen im mind 4 augen prinzip dann die abgestimmten änderungen. 
soviele signer wie wollen / mind. 3 signer nötig für autoupdate 
.. die signer testen dann die firmware bzw schauen ob überhaupt ein update angedacht war.
der admin der den speicherort der updatedateien verwaltet, hat auch nochmal ein auge drauf wann dort dateien gespeichert werden.
die vorstände der vereine haben die oberhand über die gateways/speicherort der updatedateien und  können eingreifen wenn admins mist machen.
und um den kreis zu schließen kann die community wieder die vorstände wählen.

natürlich sollten damit das funktioniert nicht allzuviele personelle überschneidungen in den verschiedenen ämtern existieren.

**Comment by bigfootDK** [comment]
Ich bin für 3 (good_signatures) von X signaturen - wobei X gerne eine hohe zahl an vertrauenswürdigen personen sein sollte.

**Comment by flemming-pr** [comment]
+1 für mehr als drei gesammt signaturen  3 aus 5 das haben wir jetzt in dem aktuellem vorschlag der site.conf

**Comment by foertel** [comment]
ich bleibe bei -1. signaturen sind für den erhalt von gleichgewicht (leider) nicht geeignet. als admin der beiden gateways lege ich euer gesigntes update dort einfach nicht hin und fertig. ;-)

abgesehen davon würde ich mich gar nicht so an der firmware aufhängen. bleibt ja jedem unbenommen, eine eigene firmware zu backen und zu verteilen und zu autoupdaten.

**Comment by Wlanfr3ak** [comment]
Dann würde Snowden hochfahren und das Update ausrollen. Toll um sowas abzustimmen.... +1 für meinen Key und alle die wollen und in der Lage sind nen Key zu generieren und mind. 3 Keys
